app/code/Magento/Swatches/Model/Plugin/EavAttribute.php

```diff
@@ -174,8 +174,10 @@ protected function prepareOptionIds(array $optionsArray)
     {
         if (isset($optionsArray['value']) && is_array($optionsArray['value'])) {
             foreach (array_keys($optionsArray['value']) as $optionId) {
-                if (isset($optionsArray['delete']) && $optionsArray['delete'][$optionId] == 1) {
-                    unset($optionsArray['value'][$optionId]);
+                if (isset($optionsArray['delete']) && isset($optionsArray['delete'][$optionId])) {
+                    if ($optionsArray['delete'][$optionId]== 1) {
```

Comment:
> empty() will consume more cpu cycle

Nope, it is same to `isset` and always faster than isset+condition. Here we avoid three conditions with only one statement.

> if $optionsArray['value'][$optionId] is set to '' (not possible here if it is set) . 

Do you mean `$optionsArray['delete'][$optionId]` ? Please provide an exact case when behavior of new and old code differs. Surely `!empty` and `isset` are not equivalent but equally suitable for this logic.